### PR TITLE
Ported emr and data

### PR DESCRIPTION
**Pull request by lasley**
This is a port of the 7.0 oemedical_emr and oemedical_emr_data modules to 8.0 (as medical_emr and medical_emr_data to match schema of existing ports). 

Mainly just a bunch of find/replace action. Everything seems to work well though! :+1:

### CODE REVIEW

File: medical_emr/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#/#############################################################################
+
+import medical_secondary_condition
```
Comment (lmignon) [inline medical_emr/models/__init__.py]: from . import medical_secondary_condition

**Comment by lmignon** [comment]
@dlasley Thank you for this PR. 
Some issues remains:
- You should fix the yml files since you have renamed oemedical -> medical.
- source code should adhere to PEP8 and PyFlakes standard

**Comment by lasley** [comment]
Hi @lmignon - No problem, I'll fix those in a sec. Sorry for my ignorance though, but what YML files? Only one I see is the TravisCI one, which doesn't seem to reference any of the changed sections.

**Comment by lasley** [comment]
Ahhhh nvm I found them. Looks like the broken Ymls were already on 8.0 in the core module. Easy enough to fix though, I'll run through the same find/replace action that I did on the other modules :smile:

**Comment by lasley** [comment]
Pyflakes and PEP standardization has been added. Pylint is yelling about relative imports, but making the recommended changes would totally break the module. Is it standard in Odoo repos to just ignore these?

Build is failing, but the error is somewhat over my head and it's working just fine on my dev box. Is there a trick here? It seems that the issue has something to do with demo data being inserted for the test, but I don't see the relevant data in the repo. Sorry, a bit new to Odoo :confused:

**Comment by lmignon** [comment]
@dlasley Thank you for all these changes.
Regarding the imports you can refer to the guidelines to see how to adapt your code https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports
In your case the content of your **init**.py files should look like for ex:

``` python
from . import oemedical_secondary_condition
from . import oemedical_operational_sector
from . import oemedical_operational_area
```

You are right, the build fails due to a prob in the demo data. The model 'medical.medicament' define a relation on product.product with 'required=True'  (https://github.com/laslabs/vertical-medical/blob/8.0/medical_emr/models/medical_medicament.py#L37) This constrain was not honored previously due to a typo in the code (https://github.com/OCA/vertical-medical/blob/08376fd140652c7121acf03bff97a9135eb98fd1/__unported__/oemedical_emr/models/oemedical_medicament.py#L38 -> `requered=True`). As you've fixed this error (s/requered/required) the demo data should be modified to add a value for the field `product_id`. For each entry in https://github.com/laslabs/vertical-medical/blob/8.0/medical_emr_data/data/WHO_list_of_essential_medicines.xml you should replace the field `name` by `product_id` as below:

``` xml
<record model="medical.medicament" id="meds_em1">
    <field name="product_id" ref="prod_em1"/>
    <field name="active_component">halothane</field>
    <field name="category" ref="em_1_1_1"/>
    <field name="pregnancy_warning">True</field>
    <field name="pregnancy_category">C</field>
    <field name="notes">Inhalation</field>
</record>
```

The the xml-id for `product_id` should be the one of the product defined in vertical-medical/medical_emr_data/data/WHO_products.xml with the same name as  in the search domain.

``` xml
<!-- field to replace -->
    <field name="name" model="product.product" search="[('name', '=', 'halothane')]"/>

<!-- product with the same name in WHO_products.xml -->        
<record model="product.product" id="prod_em1">
    <field name="name">halothane</field>
    <field name="categ_id" ref="prod_medicament_WHO"/>
    <field name="is_medicament">1</field>
    <field name="uom_id" ref="product.product_uom_unit"/>
    <field name="list_price" eval="0.0"/>
    <field name="standard_price" eval="0.0"/>
</record>

<!-- field replaced -->
    <field name="product_id" ref="prod_em1"/>
```

**Comment by lasley** [comment]
@lmignon Thanks for the detailed breakdown - the maintainer-tools repo is exactly what I needed!

We're now stuck at the issue plaguing #19 :hankey:

**Comment by lmignon** [comment]
@dlasley It seems you have replaced all the fields `name` in WHO_list_of_essential_medicines.xml with the same `<field name="product_id" ref="prod_em1"/>`.  It's not right. For each product, you should use the xmlid of the product defined in https://github.com/laslabs/vertical-medical/blob/6cc0f212b173d88f257dc522763b547f2dfe6c5e/medical_emr_data/data/WHO_products.xml with the same name as in the search domain of the replaced field `<field name="name">lidocaine + epinephrine (adrenaline)</field>`

I did not pay attention to the previous PR #19 but it's seems you're doing the same kind of changes. It might be more constructive to join your efforts with this first to finalize it.

**Comment by lasley** [comment]
@lmignon - Ohhhh that makes sense! Making the relations should be easy enough. I much prefer the look of the new Odoo API vs the old one.

And yeah I just realized we were duplicating effort once I started Googling how to fix the error I hit. I wish I would have noticed before all of this, because now we have a merge nightmare on our hands :cold_sweat: 

I should have some time to comb over the 70ish conflicts sometime tomorrow, then we can go from there.

**Comment by lmignon** [comment]
@dlasley thanks in advance :smile:

**Comment by lasley** [comment]
@lmignon - Code merged, and the slew of issues fixed. Build passing! :rocket:

**Comment by lasley** [comment]
And now it actually works!

**Comment by TheCloneMaster** [comment]
Great !!

Now that I see you going back and forth with physician and doctor, and appointment / evaluations...  I have been there too...  :(

We need to define naming conventions and project scope...

As an example, base medical module right now only enables basic patient info and appointments.  At this stage, they are so simple that calling them "evaluations" is misleading...
If you install medical_emr, then we now have medical records, and appointments now can have medical patient data and become "evaluations"....

Some customers could be only interested on appointments, and not a full EMR system, that's why I divided modules in that way...

Any ideas on how could we work on such plannings ?

Regards,

File: __unported__/oemedical_genetics/oemedical_disease_gene.py [hunk 2]
```diff
@@ -30,46 +30,48 @@ class OeMedicalDiseaseGene(orm.Model):
...
+        'patient_id': fields.many2one(
+            'oemedical.patient',
+            'Patient',
+            select=True),
+        'disease_gene': fields.many2one(
+            'oemedical.disease.gene',
+            'Disease Gene',
+            required=True),
+    }
 PatientGeneticRisk()
```
Comment (lmignon) [inline __unported__/oemedical_genetics/oemedical_disease_gene.py:60]: You can remove this line

File: __unported__/oemedical_genetics/oemedical_disease_gene.py [hunk 3]
```diff
@@ -30,46 +30,48 @@ class OeMedicalDiseaseGene(orm.Model):
...
+            'Official Symbol', size=256, required=True), 'gene_id': fields.char(
+            'Gene ID', size=256), 'long_name': fields.char(
+                'Official Long Name', size=256, required=True), 'location': fields.char(
+                    'Location', size=256, required=True, help="Locus of the chromosome"), 'chromosome': fields.char(
+                        'Affected Chromosome', size=256, required=True), 'info': fields.text(
+                            string='Information'), 'dominance': fields.selection(
+                                [
+                                    ('d', 'dominant'), ('r', 'recessive'), ], 'Dominance', select=True)}
 
 OeMedicalDiseaseGene()
```
Comment (lmignon) [inline __unported__/oemedical_genetics/oemedical_disease_gene.py:43]: You can remove this line

File: __unported__/oemedical_genetics/oemedical_disease_gene.py [hunk 4]
```diff
@@ -81,22 +83,30 @@ class FamilyDiseases(orm.Model):
...
             ('cousin', 'Cousin'),
-            ], 'Relative',
+        ], 'Relative',
             help="First degree = siblings, mother and father; second degree = "
             "Uncles, nephews and Nieces; third degree = Grandparents and cousins",
             required=True),
-            }
+    }
 
 FamilyDiseases()
```
Comment (lmignon) [inline __unported__/oemedical_genetics/oemedical_disease_gene.py:92]: You can remove this line

File: __unported__/oemedical_genetics/oemedical_disease_gene.py [hunk 5]
```diff
@@ -81,22 +83,30 @@ class FamilyDiseases(orm.Model):
...
+            'oemedical.patient.genetic.risk',
+            'patient_id',
+            'Genetic Risks'),
+        'family_history': fields.one2many(
+            'oemedical.patient.family.diseases',
+            'patient_id',
+            'Family History'),
+    }
 
 oemedicalPatient()
```
Comment (lmignon) [inline __unported__/oemedical_genetics/oemedical_disease_gene.py]: You can remove this line

File: __unported__/oemedical_genetics/oemedical_disease_gene.py [hunk 6]
```diff
@@ -81,22 +83,30 @@ class FamilyDiseases(orm.Model):
...
             help="First degree = siblings, mother and father; second degree = "
             "Uncles, nephews and Nieces; third degree = Grandparents and cousins",
             required=True),
-            }
+    }
 
 FamilyDiseases()
 
+
 class oemedicalPatient(orm.Model):
```
Comment (lmignon) [inline __unported__/oemedical_genetics/oemedical_disease_gene.py:95]: s/oemedicalPatient/OemedicalPatient

File: __unported__/oemedical_gynecology_and_obstetrics/models/oemedical_gynecology_and_obstetrics.py [hunk 7]
```diff
@@ -129,256 +175,242 @@ class PuerperiumMonitor(orm.Model):
...
+        'last_colposcopy': fields.date('Date', help="Last colposcopy", required=True),
+        'result': fields.selection([
+            ('normal', 'normal'),
+            ('abnormal', 'abnormal'),
+        ], 'result', help="Please check the lab test results if the module is installed", sort=False),
+        'comments': fields.char('Remarks'),
+    }
 
 
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline __unported__/oemedical_gynecology_and_obstetrics/models/oemedical_gynecology_and_obstetrics.py]: You can remove this line

File: __unported__/oemedical_his/models/oemedical_hospital_building.py [hunk 8]
```diff
@@ -28,12 +28,20 @@ class OeMedicalHospitalBuilding(orm.Model):
...
+        'name': fields.char(
+            size=256,
+            string='Name',
+            required=True,
+            help='Name of the building within the institution'),
+        'extra_info': fields.text(
+            string='Extra Info'),
     }
 
 OeMedicalHospitalBuilding()
```
Comment (lmignon) [inline __unported__/oemedical_his/models/oemedical_hospital_building.py]: You can remove this line

File: __unported__/oemedical_his/models/oemedical_hospital_unit.py [hunk 9]
```diff
@@ -28,12 +28,20 @@ class OeMedicalHospitalUnit(orm.Model):
...
+        'name': fields.char(
+            size=256,
+            string='Name',
+            required=True,
+            help='Name of the unit, eg Neonatal, Intensive Care, ...'),
+        'extra_info': fields.text(
+            string='Extra Info'),
     }
 
 OeMedicalHospitalUnit()
```
Comment (lmignon) [inline __unported__/oemedical_his/models/oemedical_hospital_unit.py]: You can remove this line

File: __unported__/oemedical_his/models/oemedical_surgery.py [hunk 10]
```diff
@@ -16,44 +16,63 @@
...
+            'Patient age',
+            size=3,
+            help='Patient age at the moment of the surgery. Can be estimative'),
+        'description': fields.char(
+            'Description',
+            size=128),
+        'extra_info': fields.text('Extra Info'),
+    }
+
+surgery()
```
Comment (lmignon) [inline __unported__/oemedical_his/models/oemedical_surgery.py:61]: You can remove this line

File: medical_emr/report/__init__.py [hunk 11]
```diff
@@ -18,8 +18,8 @@
 #    You should have received a copy of the GNU Affero General Public License
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
-#/#############################################################################
-import oemedical_vaccination
+# #############################################################################
 
+from . import prescription_order
 
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/report/__init__.py]: You can remove this line

File: medical_emr/models/medical_signs_and_symptoms.py [hunk 12]
```diff
@@ -18,26 +18,24 @@
...
         'sign_or_symptom': fields.selection([
             ('sign', 'Sign'),
             ('symptom', 'Symptom'),
-        ], string='Subjective / Objective',required=True),
+        ], string='Subjective / Objective', required=True),
         'comments': fields.char(size=256, string='Comments'),
     }
 
-
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_signs_and_symptoms.py]: You can remove this line

File: medical_emr/models/medical_secondary_condition.py [hunk 13]
```diff
@@ -18,22 +18,20 @@
...
+        'pathology_id': fields.many2one('medical.pathology',
                                         string='Pathology', required=True),
-        'evaluation_id': fields.many2one('oemedical.patient.evaluation',
+        'evaluation_id': fields.many2one('medical.patient.evaluation',
                                          string='Evaluation', readonly=True),
         'comments': fields.char(size=256, string='Comments'),
     }
 
-
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_secondary_condition.py]: You can remove this line

File: medical_emr/models/medical_procedure.py [hunk 14]
```diff
@@ -33,5 +32,4 @@ class OeMedicalProcedure(orm.Model):
         'name': fields.char(size=256, string='Code', required=True),
     }
 
-
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_procedure.py]: You can remove this line

File: medical_emr/models/medical_prescription_order.py [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+            'ids': ids,
+            'form': self.read(cr, uid, ids[0], context=context),
+        }
+        return {'type': 'ir.actions.report.xml',
+                'report_name': 'prescription.order',
+                'datas': datas,
+                'nodestroy': True}
+
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_prescription_order.py]: You can remove this line

File: medical_emr/models/medical_prescription_line.py [hunk 16]
```diff
@@ -0,0 +1,185 @@
...
+                                          'drug'),
+
+    }
+
+    _defaults = {
+        'prnt': True,
+
+    }
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_prescription_line.py]: You can remove this line

File: medical_emr/models/medical_prescription_line.py [hunk 17]
```diff
@@ -0,0 +1,185 @@
...
+        result = {}
+
+        # if name == 'form':
+        # result = {'value': {
+        # 'form' : medication_obj.browse(
+        #     cr, uid, medication, context = None).form.id ,
+        # } }
+        return result
+
+    # def _get_dose(self, cr, uid, ids, field_name, arg, context=None):
```
Comment (lmignon) [inline medical_emr/models/medical_prescription_line.py]: Do we need to keep these commented lines?

File: medical_emr/models/medical_patient.py [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+                                          'allergies, ...'),
+        'diseases': fields.one2many('medical.patient.disease', 'patient_id',
+                                    string='Diseases',
+                                    help='Mark if the patient has died'),
+        'ethnic_group': fields.many2one('medical.ethnicity',
+                                        string='Ethnic group', ),
+        'cod': fields.many2one('medical.pathology', string='Cause of Death', ),
+    }
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_patient.py]: You can remove this line

File: medical_emr/models/medical_pathology_group.py [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+                            help='for example MDG6 code will contain the '
+                                 'Millennium Development Goals # 6 diseases : '
+                                 'Tuberculosis, Malaria and HIV/AIDS'),
+        'name': fields.char(size=256, string='Name', required=True,
+                            translate=True, help='Group name'),
+        'desc': fields.char(size=256, string='Short Description',
+                            required=True),
+    }
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_pathology_group.py]: You can remove this line

File: medical_emr/models/medical_pathology_category.py [hunk 20]
```diff
@@ -18,27 +18,26 @@
...
+
     _constraints = [
-        (orm.Model._check_recursion, 'Error ! You cannot create recursive \n' 
-        'Category.', ['parent_id'])
+        (osv.osv._check_recursion, 'Error ! You cannot create recursive \n'
+                                   'Category.', ['parent_id'])
     ]
 
-
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_pathology_category.py]: You can remove this line

File: medical_emr/models/medical_pathology.py [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+        'gene': fields.char(size=256, string='Gene'),
+        'chromosome': fields.char(size=256, string='Affected Chromosome',
+                                  help='chromosome number'),
+    }
+
+#    _sql_constraints = [
+#        ('name_uniq', 'UNIQUE(name)', 'Name must be unique!'),
+#    ]
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_pathology.py]: You can remove this line

File: medical_emr/models/medical_pathology.py [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+                                       'belongs. Some automated processes act '
+                                       'upon the code of the group'),
+        'protein': fields.char(size=256, string='Protein involved',
+                               help='Name of the protein(s) affected'),
+        'gene': fields.char(size=256, string='Gene'),
+        'chromosome': fields.char(size=256, string='Affected Chromosome',
+                                  help='chromosome number'),
+    }
+
+#    _sql_constraints = [
```
Comment (lmignon) [inline medical_emr/models/medical_pathology.py:55]: Can we uncomment the _sql_contraints?

File: medical_emr/models/medical_operational_sector.py [hunk 23]
```diff
@@ -18,25 +18,23 @@
...
-        'name': fields.char(size=256, string='Op. Sector', required=True, 
+        'name': fields.char(size=256, string='Op. Sector', required=True,
                             help='Region included in an operational area'),
     }
     _sql_constraints = [
         ('name_uniq', 'UNIQUE(name)', 'Name must be unique!'),
     ]
 
-
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_operational_sector.py]: You can remove this line

File: medical_emr/models/medical_medication_template.py [hunk 24]
```diff
@@ -0,0 +1,104 @@
...
+        'admin_times': fields.char(size=256, string='Admin hours',
+                                   help='Suggested administration hours. '
+                                        'For example, at 08:00, 13:00 and '
+                                        '18:00 can be encoded like 08 13 18'),
+        'dose_unit': fields.many2one('product.uom', string='dose unit',
+                                     help='Unit of measure for the medication '
+                                          'to be taken'),
+    }
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_medication_template.py]: You can remove this line

File: medical_emr/models/medical_medication_dosage.py [hunk 25]
```diff
@@ -39,5 +40,4 @@ class OeMedicalMedicationDosage(orm.Model):
         ('name_uniq', 'UNIQUE(name)', 'Name must be unique!'),
     ]
 
-
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_medication_dosage.py]: You can remove this line

File: medical_emr/models/medical_medicament.py [hunk 26]
```diff
@@ -0,0 +1,101 @@
...
+        'adverse_reaction': fields.text(string='Adverse Reactions'),
+        'dosage': fields.text(string='Dosage Instructions',
+                              help='Dosage / Indications'),
+        'pregnancy': fields.text(string='Pregnancy and Lactancy',
+                                 help='Warnings for Pregnant Women'),
+        'presentation': fields.text(string='Presentation'),
+        'composition': fields.text(string='Composition', help='Components'),
+    }
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_medicament.py]: You can remove this line

File: medical_emr/models/medical_family_member.py [hunk 27]
```diff
@@ -18,21 +18,19 @@
...
+        'member': fields.many2one('res.partner', string='Member',
                                   help='Family Member Name'),
         'role': fields.char(size=256, string='Role', required=True),
-        'family_id': fields.many2one('oemedical.family', string='Family', 
+        'family_id': fields.many2one('medical.family', string='Family',
                                      help='Family code'),
     }
 
-
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_family_member.py]: You can remove this line

File: medical_emr/models/medical_family.py [hunk 28]
```diff
@@ -18,27 +18,26 @@
...
-        'members': fields.one2many('oemedical.family_member', 'family_id',
+        'members': fields.one2many('medical.family_member', 'family_id',
                                    string='Family Members', ),
     }
     _sql_constraints = [
         ('name_uniq', 'UNIQUE(name)', 'Family Code must be unique!'),
     ]
 
-
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_family.py]: You can remove this line

File: medical_emr/models/medical_ethnicity.py [hunk 29]
```diff
@@ -18,24 +18,22 @@
...
+        'name': fields.char(size=256, string='Name', required=True,
                             translate=True),
     }
     _sql_constraints = [
-        ('name_uniq', 'UNIQUE(name)', 'Name must be unique!'),
+        ('name_uniq', 'UNIQUE(name)', 'Ethnicity name must be unique!'),
     ]
 
-
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_ethnicity.py]: You can remove this line

File: medical_emr/models/medical_drug_route.py [hunk 30]
```diff
@@ -33,7 +32,7 @@ class OeMedicalDrugRoute(orm.Model):
                             translate=True),
     }
     _sql_constraints = [
-        ('name_uniq', 'UNIQUE(name)', 'Name must be unique!'),
+        ('name_uniq', 'UNIQUE(name)', 'Drug Route name must be unique!'),
     ]
 
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_drug_route.py]: You can remove this line

File: medical_emr/models/medical_drug_form.py [hunk 31]
```diff
@@ -33,7 +32,7 @@ class OeMedicalDrugForm(orm.Model):
                             translate=True),
     }
     _sql_constraints = [
-        ('name_uniq', 'UNIQUE(name)', 'Name must be unique!'),
+        ('name_uniq', 'UNIQUE(name)', 'Drug name must be unique!'),
     ]
 
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/medical_drug_form.py]: You can remove this line

File: medical_emr/models/__init__.py [hunk 32]
```diff
@@ -0,0 +1,52 @@
...
+from . import medical_patient
+from . import medical_drug_route
+from . import medical_family
+from . import medical_occupation
+from . import medical_disease_group_members
+from . import medical_medicament
+from . import medical_pathology_group
+
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/models/__init__.py]: You can remove this line

File: medical_emr/__openerp__.py [hunk 33]
```diff
@@ -136,6 +137,6 @@
 
     ],
     "active": False,
-    'installable': False,
+    "installable": True
 }
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (lmignon) [inline medical_emr/__openerp__.py]: You can remove this line

**Comment by lmignon** [comment]
@dlasley Thank you for you rebase on #19. Some additional comments (sorry for the noise). You can ignore those on files in the **unported** directory.

File: medical/models/medical_physician.py [hunk 34]
```diff
@@ -25,71 +25,120 @@
...
+        'duration': fields.selection(minutes, string='Duration'),
     }
 
 
 class MedicalPhysician(orm.Model):
     _name = 'medical.physician'
-    _inherits = {
-        'res.users': 'user_id',
-    }
+    _inherits = {'res.users': 'user_id', }
```
Comment (lmignon) [inline medical/models/medical_physician.py]: @TheCloneMaster IMO the physician should inherit from res.partner. The fact that a physician can access to the system and therefore is also a res.user is under the responsibility of the organisation which uses these addons. (one res.user = one payable user for Odoo)

File: medical_emr/models/medical_pathology.py (hunk 22 above)
Comment (lasley) [inline medical_emr/models/medical_pathology.py:55]: To be honest, no clue. @TheCloneMaster - thoughts? I'm basically a week into this, so I don't have much clue what I actually did aside from making everything work. I just needed something usable for an MVP to demo to a client :wink:

**Comment by lasley** [comment]
I ran a sed to kill all the VIM lines. ~~Mind running your bot again~~ EDIT: comments magically fixed themselves - @lmignon - is this script something I can get and run locally?

**Comment by lasley** [comment]
@TheCloneMaster - I agree that some guidelines need to be nailed down, but I think you may be thinking a little too deep. We should be using the most abstract internal terminology possible, while maybe updating the string helpers if we feel that they are better suited for that module (eg the appt vs eval scenario). 

I only breached into the med industry about a year ago, so I'm total green when it comes to the terminology. Really we probably should have used the `doctor` terminology vs. `physician`, because they could be a therapist, vet, or other. 

In terms of planning, I'm not really sure. My typical workflow is within the Atlassian tools, so I'm not really sure what is available for Github, if anything. Maybe just work within the Readme's, but that seems a little archaic? My company has a Confluence instance for Open Source dev if we wanted to prop a project there. 

@lmignon - you seem to be some sort of BDFL here, you have any thoughts?

**Comment by dlasley** [comment]
FYI I went ahead and killed the VIM/Whitespace on (allthethings) instead of just what we were working with. Figured while the automation was written, it made sense to run it everywhere.

File: medical/models/medical_physician.py (hunk 34 above)
Comment (lasley) [inline medical/models/medical_physician.py]: Agreed

File: medical_emr/models/medical_prescription_line.py (hunk 17 above)
Comment (lasley) [inline medical_emr/models/medical_prescription_line.py]: Taking the hint from dear Linus, that's why we have Git. Removing

**Comment by lmignon** [comment]
@dlasley Once again thank you for all these changes. IMO, two things are key stoppers for merging this PR in this state.
1 MedicalPhysician inheriting from res_users
2 The lack of consistency in the way some fields are named. 
At least, The first one is blocking.

**Comment by lasley** [comment]
@lmignon - Inheritance modified. The lack of consistency was present from the beginning, so I think maybe we should do that in a different PR. I already breached scope of the original 8.0 port PR by rebasing, so I'd rather not expand the scope again if possible.

**Comment by lmignon** [comment]
@lasley @TheCloneMaster IMO, this PR can be merged as it if we agree that renaming some fields in an other PR will not be blocked.

**Comment by lasley** [comment]
Sounds like a plan to me! :+1:

**Comment by lmignon** [comment]
@lasley @TheCloneMaster  @nhomar  Since this branch is probably not use at all, IMO it would be nice to create a 8.0-incubator branch and merge this one into the new one. The incubator branch could be used until it has reached a sufficient degree of maturity.The goal is to be able to "freely" rework models to provide a solid and clean base for further developments. We are in a analysis phase for one of our client and we would like to take benefits of this new project to revive and expand vertical_medical.

**Comment by TheCloneMaster** [comment]
Sounds good for me.  I agree that having doctors as users could be a
problem in terms on money because of cost per user...

Question remains...  How can we set at least some design goals and
guidelines, naming conventions, ...  ?

**Comment by lasley** [comment]
Making a release branch is probably a good course of action here. I am also in an eval stage, with the plan relying heavily on this library. I'm still in the process of checking that we can meet HIPAA compliance with this setup, but I'm thinking it's possible with proper server configs. 

In terms of the design goals, my best idea is Github Wiki :sob:

**Comment by lasley** [comment]
Hey @lmignon & @TheCloneMaster (& anyone else involved in this project :wink: ) - 

Short of ideas for managing all this, I setup Confluence and JIRA instances for this project at https://vertical-medical.atlassian.net (wiki https://vertical-medical.atlassian.net/wiki/dashboard.action )

I figure we should just mirror the repo folders with JIRA projects, then manage the forward development there. We can use Confluence for the wiki-ish things, as well as the Gliffy plugin for diagrams. 

Honestly, this is just the workflow I use in my day-to-day (except I use Stash too, but meh). If anyone else has a better idea, I'm all ears; but Github sucks for this IMO. 

Also, should I close this PR?

**Comment by sbidoul** [comment]
@lasley I think it's important to keep consistent tooling across OCA projects because introducing new tooling would raise the entry barrier to other OCA contributors. Moreover many current and upcoming OCA tools work across repositories and expect consistent use of github. So I'd rather avoid going to JIRA & co for now. Github issues and wiki should be good enough to start.

I'm in favor of the 8.0-incubator branch too, as it gives a signal to the community that this is still not 100% stable and will facilitate working with smaller PR's.

I'll create the branch right now from this PR and close the PR.

**Comment by sbidoul** [comment]
8.0-incubator branch created. IMO the lifespan of this branch should be as short as possible and as soon as people around here reach an agreement it should be merged in 8.0.

**Comment by lasley** [comment]
Woot thanks for the guidance @sbidoul . I agree that we should kill the incubator as quickly as possible, but this will help us build the momentum we need to get this viable for prime time somewhat quickly. 

@TheCloneMaster @lmignon -

I started a HIPAA Compliance page in the Wiki, which will be my focus for the beginning part of this. As it stands, this library cannot be used in a US healthcare environment without significant modifications to the server. I would like for the library itself to contain the security necessary to align with HIPAA on a SaaS Odoo Deploy. 

Some of the things I noted may already be implemented, and I just don't know about them. I'm only about a month into Odoo, so I'll probably ask for some expert advice once I nail down a solid list of requirements :wink: 

I don't have knowledge regarding healthcare standards abroad, so I am open to any suggestions from anyone that does. I'm sure there are laws/standards parallel to HIPAA in basically any first world country, so it would be nice to get somewhat of an all-inclusive thing going.

**Comment by lmignon** [comment]
@lasley I'm on vacation until monday 19. I'll read the wiki once I'll be back to the work.

**Comment by Olivier-LAURENT** [comment]
Hi @lasley 
Do you plan to continue the refactoring of medical data models?
At least, the name of the most of relation fields m2o and o2m must be reviewed to respect usual namming conventions. This task is the minimal condition to allow to merge Incubator branch into 8.0.
If yes, thx in advance for this work. What's your approximative planning?
If not, we could make the part needed for our project but in this case some unnecessary models will be moved back to `__unported__` allowing others to achieve or just complete this migration job.
The goal is to not block for a long the merge of incubator. 
Keep us in touch. Thx a lot.

**Comment by lasley** [comment]
Hi @Olivier-LAURENT 

Yeah I've been meaning to start pruning these models. Is there maybe a doc (or maybe just some solid new API code) you can point me towards in order to identify the naming conventions? I'm fairly new to Odoo, and docs regarding this stuff seem pretty slim. 

In terms of planning, I can have them refactored in the next week or so assuming I know what to do. I've honestly just been biding my time playing in the code to try and get a hang of the conventions, but it seems there's a lot of old API conventions left over so I'm getting bogged down :|

**Comment by lmignon** [comment]
@lasley Guidelines are available here https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md

**Comment by lasley** [comment]
Awesome, I've been looking for Odoo specific ones; makes sense the OCA would have a solid set. Thanks @lmignon, sorry for my newb status!

I'll have some time to rock these out within the next week or so :smile:

**Comment by lasley** [comment]
This is going to affect my open pull request #24, should I just kill it and open a new one when this is ready?

**Comment by Olivier-LAURENT** [comment]
Hi @lasley 
Regarding guidelines pointed by @lmignon, the most important is:
- the data model of the database, i.e., the name of models (tables) and fields, and if possible
- the name of the xml_id of views, actions, menus, groups ...

All other kinds of rules can be improved progressively later without breaking effective deployments. In this way, typically, your PR #24 seems to be not affected.
I also suggest before refactoring medical and medical_emr  to review and merge first the PRs #27 and #31 (in this order). Some such refactorings are already proposed there by @JonathanNEMRY.

**Comment by lasley** [comment]
Hi @Olivier-LAURENT 

Got it, thanks for clarifying. I have read the guidelines, and started the upgrade. I'll focus on the model and xml_id naming so we stop blocking PRs. 

Thanks for pointing out the #27 and #31, I'll merge them into my working branch. I'm hoping to have an updated PR for us this week.

**Comment by lmignon** [comment]
@lasley Don't merge #27 and #31 in you branch. It is better to rebase your branch on #31 . #31 is already based on #27.

**Comment by lasley** [comment]
@lmignon - I like it, much cleaner. Thanks for all the tips, everyone

**Comment by lasley** [comment]
Hi @lmignon and @Olivier-LAURENT - Would one of you mind taking a look at the security to make sure I did it right? The docs were a bit unclear when it came to CSV -- https://github.com/laslabs/vertical-medical/blob/8.0-incubator_naming-to-oca-specs/medical_emr/security/ir.model.access.csv

**Comment by lfend** [comment]
Hi,

I took a look at your security file. Until now, based on what I've seen in other modules (and on my own experience):
- The string of the `id` column is formatted like this: `access_name_of_the_model_X` with 'X' being a element to differentiate each line (like user/manager for example). Based on that, your first line would be `access_medical_secondary_condition_user` for example.
- The string for the `name` column is a pointed notation of the 'id' string, without the "access". So, again with the first line example:`'medical.secondary.condition.user`.

I take personally this as a best practice. Btw, I didn't take a look the access rights by themselves, I just started to get involve in the vertical-medical.

**Comment by lasley** [comment]
Awesome thanks @lfend - I'll update accordingly.

**Pull request closed without merging**